woocommerce/woocommerce
# Add Tags to product editor

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces `Tags` handling to the blocks editor, enhancing the user's experience while handling products. Users can now create, add, remove, and search for existing tags. This process mirrors the way we currently manage `Categories`.

![Screenshot 2023-09-01 at 09 28 53](https://github.com/woocommerce/woocommerce/assets/1314156/450db233-c72a-4622-8304-e693a44955c3)


Closes #39126.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the product blocks editor under WooCommerce -> Settings -> Advanced -> Features.
2. Navigate to Products > Add new.
3. Press `Organization` and select the Tags dropdown menu.
4. You'll see the tags that you already had.
5. Press `Create new` and add the name of a new tag.
6. After a brief moment, the new tag should appear in both the dropdown menu and the input field.

![Screen Capture on 2023-09-01 at 10-06-24](https://github.com/woocommerce/woocommerce/assets/1314156/65b54e1c-eb73-44bc-9549-318c6fc49369)

7. While still within the Tags selector, start typing the name of a new tag.
8. Press `Create [tag name that you typed]`.
9. After a few seconds, the new tag should be added to the selector.

![Screen Capture on 2023-09-01 at 10-11-47](https://github.com/woocommerce/woocommerce/assets/1314156/3ac4fd97-c84c-402e-9f7d-64fc2c76b046)

10. To ensure proper functionality, verify that the tags are saved correctly, both individually and when assigned to a product.

// cc @jarekmorawski I've mirrored the `Categories` selector behavior. Could you check whether the tags selector looks and functions as expected?



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
